app/main.py

```diff
@@ -1,34 +1,90 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row, column, is_alive=True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
+    def __init__(self, start: tuple, end: tuple, is_drowned=False) -> None:
         # Create decks and save them to a list `self.decks`
-        pass
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = []
+        self.create_ship()
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def create_ship(self) -> None:
+        if self.start[0] == self.end[0] and self.start[1] == self.end[1]:
+            self.decks.append(Deck(self.end[0], self.end[1]))
+        elif self.start[0] == self.end[0]:
+            for coords in range(self.start[1], self.end[1] + 1):
+                self.decks.append(Deck(self.start[0], coords))
+        elif self.start[1] == self.end[1]:
+            for coords in range(self.start[0], self.end[0] + 1):
+                self.decks.append(Deck(coords, self.start[1]))
 
-    def fire(self, row, column):
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+
+    def fire(self, row: int, column: int) -> None:
         # Change the `is_alive` status of the deck
         # And update the `is_drowned` value if it's needed
-        pass
+        self.get_deck(row, column).is_alive = False
+        self.check_if_drowned()
+
+    def check_if_drowned(self) -> None:
+        for deck in self.decks:
+            if deck.is_alive:
+                self.is_drowned = False
+                return
+            self.is_drowned = True
```

Comment:
can you try to do the same with `any()` and list comprehension? That is not a mistake, your solution is good. But I think that it will be more interesting

Reply:
WDYT?

Reply:
nice idea. That would really improve code style, i guess.